enhancements/security/scc-run-as-any.md

```diff
@@ -0,0 +1,134 @@
+---
+title: scc-run-as-any
+authors:
+  - "@alanconway"
+reviewers:
+  - "@eparis"
+  - "@bparees"
+  - "@jcantrill"
+approvers:
+  - "@eparis"
+api-approvers:
+  - None
+creation-date: 2023-05-17
+last-updated:
+tracking-link:
+  - TBD
+---
+
+# Consistent handling for SecurityContextConstraint RunAsAny strategies
+
+## Summary
+
+SecurityContextConstraints serve two distinct functions:
+
+1. They _constrain_ Pod configurations that are allowed to run.
+2. They _modify_ Pod configurations with values from outside the Pod based on strategies in the SCC.
+
+From [the docs][scc]:
+> The admission controller is aware of certain conditions in the security context constraints (SCCs) that trigger it to look up pre-allocated values from a namespace and populate the SC before processing the pod.
+
+SCC strategies are ordered by "restrictiveness": if a Pod runs under SCC X, then it MUST run under any SCC that is _less restrictive_ than X.
+This ordering is critical to correctly matching SCCs to Pods, and must be clear an consistent.
+
+The `RunAsAny` strategy is documented as being the _least restrictive_ `runAsUser` strategy.
+In other words, a Pod that runs under some other `runAsUser` strategy must also run under `RunAsAny`
```

Comment:
> you cannot test a Pod configuration against an SCC, and assume it will work under all matching and "less restrictive" SCCs.

Agreed, and i guess i also have to defer to the SCC experts here, but my view is that that was never a goal/claim/intent of SCCs.

The SCC claim would be the a pod that is *admitted* (oc create succeeds) under one SCC, must also be admitted under a less restrictive SCC (though if you have 2 SCCs, one which allows only permission A, and one which allows unrelated permissions B+C but not permission A, which one of those is "less restrictive"?  This is something else that isn't really well defined in the SCC docs...it just says the most restrictive SCC that the user has access to and which can admit the pod, will be chosen(assuming equal priorities).  I don't think it makes any claims that the runtime behavior will be the same under either SCC.

I agree that's subtle and confusing to users and at a minimum should be documented very clearly.